miden-lib/asm/kernels/transaction/lib/account.masm

```diff
@@ -433,16 +412,20 @@ export.set_item
     emit.ACCOUNT_STORAGE_BEFORE_SET_ITEM_EVENT
     # => [index, V']
 
-    # check if storage type is slot
-    dup exec.memory::get_acct_storage_slot_type_data u32split drop
-    # => [slot_type, index, V', 0, 0, 0]
+    # check that index is in bounds
+    dup exec.memory::get_num_storage_slots lt assert.err=ERR_STORAGE_SLOT_INDEX_OUT_OF_BOUNDS
+    # => [index, V']
```

Comment:
Similar to the last comment, don't we usually call `apply_storage_offset` before calling this procedure and so this check should have already been done?

Reply:
answered above